### ISSUE TITLE
Re-re-organize _set_checksum_address

### ISSUE THREAD

**Issue by jMyles** [nucypher/nucypher#2228]
I think the general organization of this method is confusing. The conditional hierarchy currently is to first consider if the input is None or not, and then it has to check if it's federated or not. Perhaps it'd better to switch the order:
```python
def _set_checksum_address(self, checksum_address=None):

    if self.federated_only:
        if checksum_address is not None:
            try:
                self._set_checksum_address()  # type: str
            except NoSigningPower:
                self._checksum_address = NO_BLOCKCHAIN_CONNECTION
            if checksum_address:
                # We'll take a checksum address, as long as it matches their signing key
                if not checksum_address == self.checksum_address:
                    error = "Federated-only Characters derive their address from their Signing key; got {} instead."
                    raise self.SuspiciousActivity(error.format(checksum_address))

        verifying_key = self.public_keys(SigningPower)
        uncompressed_bytes = verifying_key.to_bytes(is_compressed=False)
        without_prefix = uncompressed_bytes[1:]
        verifying_key_as_eth_key = EthKeyAPI.PublicKey(without_prefix)
        public_address = verifying_key_as_eth_key.to_checksum_address()

    else:  # Decentralized
        if checksum_address is not None:
            self._checksum_address = checksum_address  # TODO: Check that this matches TransactingPower
        try:
            # TODO: Some circular logic here if we haven't set the canonical address.
            public_address = to_checksum_address(self.canonical_public_address)
        except TypeError:
            public_address = NO_BLOCKCHAIN_CONNECTION
            # raise TypeError("You can't use a decentralized character without a _checksum_address.")
        except NotImplementedError:
            raise TypeError(
                "You can't use a plain Character in federated mode - you need to implement ether_address.")  # TODO: update comment

    self._checksum_address = public_address
```

_Originally posted by @cygnusv in https://github.com/nucypher/nucypher/pull/1741#discussion_r482197987_

**Comment by jMyles** [comment]
And uhh, #1995

**Comment by KPrasch** [comment]
Closed in #2536
